libraries/AP_HAL_ChibiOS/hwdef/3DR-L431/hwdef-bl.inc

```diff
@@ -0,0 +1,60 @@
+# HW definition file for 3DR CAN Nodes (STM32L431)
+
+# MCU class and specific type
+MCU STM32L431 STM32L431xx
+
+# use LSE clock source
+OSCILLATOR_HZ 32768
+
+# board ID. See Tools/AP_Bootloader/board_types.txt
+APJ_BOARD_ID AP_HW_3DR_L431
+
```

Comment:
Can't you just omit this?  i.e. you will only ever create boards which inherit from this, so the base board doesn't need an ID

If you can get away with this, then leave a comment in saying that the inheriting boards will set the board id.